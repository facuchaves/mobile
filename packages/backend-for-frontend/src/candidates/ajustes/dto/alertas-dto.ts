import { ApiModelProperty } from '@nestjs/swagger'

class FrecuenciaNotificacion {
  id: number
  nombre: string
}
class AlertaLocalizacion {
  @ApiModelProperty({ required: false })
  localizacionesId?: string[]
  @ApiModelProperty({ required: false })
  tipo?: string[]
}
class Pais {
  id?: number
  isoCode?: string
  nombre?: string
}
class SubArea {
  id: number
  nombre: string
}
class TipoTrabajo {
  id: number
  nombre: string
}

export class AlertaResCore {
  ciudad?: string
  criterios?: string[]
  frecuenciaNotificacion?: FrecuenciaNotificacion
  id?: number
  localizacion: AlertaLocalizacion
  pais?: Pais
  query?: string
  salarioMinimo?: number
  subAreas?: SubArea[]
  tipoTrabajo?: TipoTrabajo
}

export class AlertaPostCore {
  @ApiModelProperty({ required: true })
  areaId: number
  @ApiModelProperty({ required: false })
  ciudad?: string
  @ApiModelProperty({ required: false })
  frecuenciaNotificacionId?: number
  @ApiModelProperty({ required: false })
  localizacion: AlertaLocalizacion
  @ApiModelProperty({ required: true })
  paisId?: number
  @ApiModelProperty({ required: false })
  query?: string
  @ApiModelProperty({ required: false })
  salarioMinimo?: number
  @ApiModelProperty({ required: true })
  subAreasId?: number
  @ApiModelProperty({ required: true })
  tipoTrabajoId?: number
}
