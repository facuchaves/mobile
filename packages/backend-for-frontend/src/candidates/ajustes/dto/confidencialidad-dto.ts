import { ApiModelProperty } from '@nestjs/swagger'

export class Confidencialidad {
  @ApiModelProperty({ required: false })
  tipo: string
}
