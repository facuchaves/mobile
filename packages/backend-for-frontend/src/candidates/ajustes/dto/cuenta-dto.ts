import { ApiModelProperty } from '@nestjs/swagger'

export class Cuenta {
  postulable: boolean
  mailValidado: boolean
  bloqueada: boolean
  fechaAlta: string
  fechaUltimoAcceso: string
  fechaAltaCurriculum: string
  cuentasAosciadasDeTerceros: string[]
}

export class CuentaDelete {
  @ApiModelProperty({ required: true })
  motivo: string
  @ApiModelProperty({ required: true })
  password: string
}
