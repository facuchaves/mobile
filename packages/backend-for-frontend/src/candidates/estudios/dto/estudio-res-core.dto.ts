import { ApiModelProperty } from '@nestjs/swagger'

export class Estudio {
  id?: number
  titulo?: string
  areaEstudio?: idNombre
  cantidadMateriasAprobadas?: number
  cantidadMateriasTotal?: number
  estadoEstudio?: idNombre
  fechaFin?: string
  fechaInicio?: string
  institucionEducativa?: InstitucionEducativa
  pais?: Pais
  promedio?: string
  rangoPromedio?: idNombre
  tipoEstudio?: idNombre
}

export class idNombre {
  @ApiModelProperty({ required: false })
  id: number

  @ApiModelProperty({ required: false })
  nombre: string
}

class InstitucionEducativa {
  @ApiModelProperty({ required: false })
  alias: string[]

  @ApiModelProperty({ required: false })
  id: number

  @ApiModelProperty({ required: false })
  nombre: string
}

export class Pais {
  @ApiModelProperty({ required: false })
  id: number

  @ApiModelProperty({ required: false })
  nombre: string

  @ApiModelProperty({ required: false })
  isoCode: string
}

export const estudioMock = {
  titulo: 'Ingeniero en Informatica',
  fechaInicio: '01-01-2002',
  fechaFin: '01-05-2003',
  promedio: null,
  cantidadMateriasTotal: 40,
  cantidadMateriasAprobadas: 4,
  institucionEducativa: {
    id: 203,
    nombre: 'Universidad de Buenos Aires',
    alias: null,
  },
  rangoPromedio: null,
  id: 75668500,
  pais: {
    id: 1,
    nombre: 'Argentina',
    isoCode: 'ar',
  },
  areaEstudio: {
    id: 30,
    nombre: 'Ing. Inform√°tica',
  },
  estadoEstudio: {
    id: 3,
    nombre: 'Abandonado',
  },
  tipoEstudio: {
    id: 3,
    nombre: 'Universitario',
  },
}
