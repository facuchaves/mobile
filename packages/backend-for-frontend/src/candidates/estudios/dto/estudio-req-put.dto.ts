import { ApiModelProperty } from '@nestjs/swagger'
import { idNombre, Pais } from './estudio-res-core.dto'

export class EstudioPut {
  @ApiModelProperty({ required: false })
  titulo?: string

  @ApiModelProperty({ required: false })
  areaEstudio?: idNombre

  @ApiModelProperty({ required: false })
  cantidadMateriasAprobadas?: number

  @ApiModelProperty({ required: false })
  cantidadMateriasTotal?: number

  @ApiModelProperty({ required: false })
  estadoEstudio?: idNombre

  @ApiModelProperty({ required: false })
  fechaFin?: string

  @ApiModelProperty({ required: false })
  fechaInicio?: string

  @ApiModelProperty({ required: false })
  institucionEducativa?: idNombre & { alias: string[] }

  @ApiModelProperty({ required: false })
  pais?: Pais

  @ApiModelProperty({ required: false })
  promedio?: string

  @ApiModelProperty({ required: false })
  rangoPromedio?: idNombre

  @ApiModelProperty({ required: false })
  tipoEstudio?: idNombre
}
