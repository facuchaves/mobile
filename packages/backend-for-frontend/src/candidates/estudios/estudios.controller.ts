import { Controller, Get, Post, Put, Delete, Param, Body, UseInterceptors, CacheInterceptor } from '@nestjs/common'
import { EstudiosService } from './estudios.service'
import { ApiUseTags, ApiResponse } from '@nestjs/swagger'
import { Context } from '../../decorators/request-context.decorator'
import { RequestContext } from '../../models/request-context'
import { EstudioPut } from './dto/estudio-req-put.dto'

@ApiUseTags('postulante-estudios - candidate-estudios')
@Controller('estudios')
export class EstudiosController {
  constructor(private readonly service: EstudiosService) {}

  @Get('/:estudioId')
  async getEstudios(@Context() context: RequestContext, @Param('estudioId') estudioId: number): Promise<any> {
    return this.service.getEstudio(context, estudioId)
  }

  @Put('/:estudioId')
  async PutEstudios(
    @Context() context: RequestContext,
    @Param('estudioId') estudioId: number,
    @Body() estudioData?: EstudioPut
  ): Promise<any> {
    return this.service.putEstudio(context, estudioId, estudioData)
  }

  @Delete('/:estudioId')
  async delEstudios(@Context() context: RequestContext, @Param('estudioId') estudioId: number): Promise<any> {
    return this.service.delEstudio(context, estudioId)
  }
}
