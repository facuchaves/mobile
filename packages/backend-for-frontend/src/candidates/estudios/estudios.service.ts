import { Injectable } from '@nestjs/common'
import { RequestContext } from '../../models/request-context'
import { CandidatesCoreApiService } from '../candidates-core-api/candidates-core-api.service'
import { Estudio } from './dto/estudio-res-core.dto'
import { EstudioPut } from './dto/estudio-req-put.dto'

@Injectable()
export class EstudiosService {
  constructor(private readonly candidatesCoreApiService: CandidatesCoreApiService) {}

  async getEstudio(context: RequestContext, estudioId: number): Promise<Estudio> {
    return this.candidatesCoreApiService.get<Estudio>(context, `/v0/postulantes/curriculum/estudios/${estudioId}`)
  }

  async delEstudio(context: RequestContext, estudioId: number) {
    return this.candidatesCoreApiService.delete<any>(context, `/v0/postulantes/curriculum/estudios/${estudioId}`)
  }

  async putEstudio(context: RequestContext, estudioId: number, bodyData: EstudioPut) {
    return this.candidatesCoreApiService.put<any>(context, `/v0/postulantes/curriculum/estudios/${estudioId}`, bodyData)
  }
}
