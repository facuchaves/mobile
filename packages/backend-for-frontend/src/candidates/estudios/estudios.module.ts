import { Module, CacheModule } from '@nestjs/common'
import { EstudiosController } from './estudios.controller'
import { EstudiosService } from './estudios.service'
import { RedisModule } from '../../cache/cache-redis/redis.module'
import { CandidatesCoreApiModule } from '../candidates-core-api/candidates-core-api.module'
import { ApplicationCoreApiModule } from '../../application-core-api/application-core-api.module'

@Module({
  imports: [RedisModule, CandidatesCoreApiModule, ApplicationCoreApiModule],
  controllers: [EstudiosController],
  providers: [EstudiosService],
})
export class EstudiosModule {}
