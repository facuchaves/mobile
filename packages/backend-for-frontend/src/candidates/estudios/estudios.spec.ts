import { EstudiosService } from './estudios.service'
import { RequestContext } from '../../models/request-context'
import { CandidatesCoreApiService } from '../candidates-core-api/candidates-core-api.service'
import { estudioMock } from './dto/estudio-res-core.dto'

describe('Curriculum', () => {
  let service
  let candidatesCoreApiService
  let spy
  beforeEach(async () => {
    candidatesCoreApiService = ({
      get: () => {},
      delete: () => {},
      put: () => {},
      post: () => {},
    } as unknown) as CandidatesCoreApiService
    service = new EstudiosService(candidatesCoreApiService)
    spy = null
  })

  afterEach(async () => {
    if (spy) spy.mockClear()
  })

  it('should get estudio data', async () => {
    const contextMocked = ({
      isCandidateLogged: () => true,
    } as unknown) as RequestContext
    spy = jest.spyOn(candidatesCoreApiService, 'get').mockImplementation(() => Promise.resolve(estudioMock))
    const result = await service.getEstudio(contextMocked, 1)
    expect(result).toMatchSnapshot()
  })

  it('should delete estudio data', async () => {
    const contextMocked = ({
      isCandidateLogged: () => true,
    } as unknown) as RequestContext
    spy = jest.spyOn(candidatesCoreApiService, 'delete').mockImplementation(() => Promise.resolve(null))
    const result = await service.delEstudio(contextMocked, 1)
    expect(result).toMatchSnapshot()
  })

  it('should put estudio data', async () => {
    const contextMocked = ({
      isCandidateLogged: () => true,
    } as unknown) as RequestContext
    spy = jest.spyOn(candidatesCoreApiService, 'put').mockImplementation(() => Promise.resolve(null))
    const result = await service.putEstudio(contextMocked, 1, {})
    expect(result).toMatchSnapshot()
  })
})
