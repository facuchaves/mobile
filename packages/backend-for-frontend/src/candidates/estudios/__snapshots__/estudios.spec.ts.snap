// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Curriculum should delete estudio data 1`] = `null`;

exports[`Curriculum should get estudio data 1`] = `
Object {
  "areaEstudio": Object {
    "id": 30,
    "nombre": "Ing. Inform√°tica",
  },
  "cantidadMateriasAprobadas": 4,
  "cantidadMateriasTotal": 40,
  "estadoEstudio": Object {
    "id": 3,
    "nombre": "Abandonado",
  },
  "fechaFin": "01-05-2003",
  "fechaInicio": "01-01-2002",
  "id": 75668500,
  "institucionEducativa": Object {
    "alias": null,
    "id": 203,
    "nombre": "Universidad de Buenos Aires",
  },
  "pais": Object {
    "id": 1,
    "isoCode": "ar",
    "nombre": "Argentina",
  },
  "promedio": null,
  "rangoPromedio": null,
  "tipoEstudio": Object {
    "id": 3,
    "nombre": "Universitario",
  },
  "titulo": "Ingeniero en Informatica",
}
`;

exports[`Curriculum should put estudio data 1`] = `null`;
